### PR TITLE
add group context

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: ApiBundle/Context/GroupContext.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace OpenOrchestra\ApiBundle\Context;
+
+/**
+ * Class GroupContext
```
Comment (juchi) [inline ApiBundle/Context/GroupContext.php]: wouldn't it be a good starting point for some code documentation as discussed some time ago ?

File: ApiBundle/EventSubscriber/AbstractSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return boolean
+     */
+    protected function isApiRequest(Request $request)
+    {
+        return 0 === strpos($request->get('_route'), 'open_orchestra_api');
```
Comment (itkg-ngilain) [inline ApiBundle/EventSubscriber/AbstractSubscriber.php]: We get a problem here. Not only on this PR but generally with that test (it is also used in the serialiser subscriber) :
to extend the api in others bundles (generic bundles or app bundles), api routes must be prefixed by open_orchestra_api, which is not logical in that cases.
Can you find a more generic test ?

File: ApiBundle/EventSubscriber/AbstractSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        if (! $event->isMasterRequest()) {
+            return false;
+        }
+
+        if (!$this->isApiRequest($event->getRequest())) {
+            return false;
+        }
+
+        return true;
```
Comment (itkg-rboessel) [inline ApiBundle/EventSubscriber/AbstractSubscriber.php]: return $event->isMasterRequest() && $this->isApiRequest($event->getRequest());

File: ApiBundle/Context/GroupContext.php (hunk 1 above)
Comment (nicolasThal) [inline ApiBundle/Context/GroupContext.php]: yes, good point :)

**Comment by nicolasThal** [comment]
@juchi @itkg-rboessel update done

**Comment by itkg-rboessel** [comment]
:+1:

File: ApiBundle/EventSubscriber/AbstractSubscriber.php (hunk 2 above)
Comment (nicolasThal) [inline ApiBundle/EventSubscriber/AbstractSubscriber.php]: I agree with you, but this discussion should be more wide than just on this pr.

I will add a ticket for the ceremonie of tomorrow, is it ok for you?

**Comment by juchi** [comment]
:+1:

File: ApiBundle/EventSubscriber/AbstractSubscriber.php (hunk 2 above)
Comment (itkg-ngilain) [inline ApiBundle/EventSubscriber/AbstractSubscriber.php]: I had this one : https://trello.com/c/Z2NZGPOL/883-code-on-check-par-endroit-si-la-requete-est-une-requete-d-api-en-testant-que-le-prefixe-de-la-route-est-open-orchestra-api-ce-qu

**Comment by itkg-ngilain** [comment]
:+1:

**Pull request merged**